tests/5.1/teams/test_target_teams_thread_limit.c

```diff
@@ -0,0 +1,65 @@
+//===---------------- test_target_teams_thread_limit.c -----------------------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+//
+// This test uses the thread_limit clause on the teams construct. Specifically
+// testing if a thread_limit from a above target construct properly carries
+// down to the nested teams construct, as if it were directly on the construct
+// as defined in the spec. The test validates that only the specified 
+// threads are created by summing a shared variable across all threads 
+// (and teams). If the threads are correctedly limited this should produce the 
+// expected value. Additional warnings are sent if specific issues occur.
+//
+////===----------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int main() {
+	int errors[OMPVV_NUM_THREADS_DEVICE/OMPVV_NUM_TEAMS_DEVICE];
+	int num_teams = 0;
+        int sum_errors = 0;
+        int i;
+	int testing_thread_limit = OMPVV_NUM_THREADS_DEVICE/OMPVV_NUM_TEAMS_DEVICE;
+
+	OMPVV_TEST_OFFLOADING;
+      
+
+         for (i = 0; i<testing_thread_limit; i++){
```

Comment:
I think this should be:
```suggestion
         for (i = 0; i<OMPVV_NUM_TEAMS_DEVICE; i++){
```
as later it is iterated over `num_teams`